pageserver/src/layered_repository/timeline.rs

```diff
@@ -1808,7 +1808,7 @@ impl Timeline {
                         break;
                     }
                     key_values_total_size += next_size;
-                    if key_values_total_size > target_file_size {
+                    if key_values_total_size > target_file_size && lsn != next_lsn {
```

Comment:
> Please, add a comment. This code is far from trivial.
> 
> Also, should'n we check lsn before we add 'next_size' in the line above? If I read it correctly, we count same layer size twice.

I have added comment.

Concerning performing check before adding `next_size`,  I believe that current code is correct.
We are going to split layer when it becomes larger than specified threshold. So if adding next key-value pair cause layer overflow then we should split it. We do not count the same key size twice, because once we exit the loop, we reset  `key_values_total_size` variable by assigning `next_key_size` (size of key-value pair which was rejected. So increment of `key_values_total_size`  above is overwritten.